mayastor/src/lvm/lvm.rs

```diff
@@ -0,0 +1,285 @@
+                disks: disks.to_owned(),
+                pooltype: PoolType::Lvm.into(),
+            }).await?,
+            _ => return Err(e)
+        }
+    };
+    info!("The pool '{}' has been created on {:?}", name, disks);
+    Ok(pool)
+}
+
+pub async fn import_lvm_vg(req: CreatePoolRequest) -> Result<Pool, Error> {
+    let pools = list_lvm_vg().await?;
+    match pools.iter().find(|p| p.name == req.name) {
+        Some(p) => {
+            if p.disks == req.disks {
+                return Ok(Pool{
+                    name: p.name.as_str().to_owned(),
+                    disks: p.disks.as_slice().to_owned(),
+                    capacity: p.capacity,
+                    used: p.used,
+                    pooltype: PoolType::Lvm.into(),
+                    state: PoolState::PoolOnline.into(),
+                });
+            }
+            Err(Error::ImportErr {err: format!("lvm vg {} disks not match", req.name)})
+        }
+        None => Err(Error::NotFound {err: format!("lvm volume {} not found", req.name)}),
+    }
+}
+
+pub async fn create_lvm_vg(req: CreatePoolRequest) -> Result<Pool, Error> {
+    let output = Command::new(PVCREATE_COMMAND)
+        .args(&req.disks)
+        .output()
+        .await
+        .expect("failed to execute pv_create");
+    if !output.status.success() {
+        let msg = match String::from_utf8(output.stderr){
+            Ok(s) => s,
+            Err(_) => "failed to parse stderr for pv_create".to_string(),
+        };
+        return Err(Error::FailedExec{err: msg})
+    }
+
+    let output = Command::new(VGCREATE_COMMAND)
+        .arg(req.name.as_str())
+        .args(&req.disks)
+        .arg("--addtag=mayastor")
+        .output()
+        .await
+        .expect("failed to execute vg_create");
+    if !output.status.success() {
+        let msg = match String::from_utf8(output.stderr){
+            Ok(s) => s,
+            Err(_) => "failed to execute vg_create".to_string(),
+        };
+        return Err(Error::FailedExec{err: msg.to_string()})
+    }
+
+    get_lvm_vg(req.name).await
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+struct VGListReport {
+    report: Vec<VGList>
+}
+#[derive(Debug, Serialize, Deserialize)]
+struct VGList {
+    vg: Vec<VGName>
+}
+#[derive(Debug, Serialize, Deserialize)]
+struct VGName {
+    vg_name: String
+}
+pub async fn list_lvm_vg() -> Result<Vec<Pool>, Error> {
+    let output = Command::new(VGS_COMMAND)
+        .args(&["--options=vg_name", "--reportformat=json"])
+        .arg("@mayastor")
+        .output()
+        .await
+        .expect("failed to execute vgs");
+    if !output.status.success() {
+        let msg = match String::from_utf8(output.stderr){
+            Ok(s) => s,
+            Err(_) => "failed to execute vgs".to_string(),
+        };
+        return Err(Error::FailedExec{err: msg.to_string()})
+    }
+
+    let r: VGListReport = match serde_json::from_slice(output.stdout.as_slice()){
```

Comment:
Descriptive names for variables please.

```suggestion
    let report: VGListReport = match serde_json::from_slice(output.stdout.as_slice()){
```